### ISSUE TITLE
DatadogTrackingHttpClient throws _TypeError if request URL is matched ignoreUrlPatterns

### ISSUE THREAD

**Issue by ronnnnn** [DataDog/dd-sdk-flutter#589]
### Describe the bug

`DatadogTrackingHttpClient` throws `_TypeError` if request URL is matched `ignoreUrlPatterns` and `innerClient` throws error  (like `SocketException`).

### Reproduction steps

1. enable Http Tracking with ignoreUrlPatterns
2. cause network problem (like no network connection, airplane mode, ...)
3. request to Url set on ignoreUrlPatterns

```dart
final datadogConfig = DatadogConfiguration(
  // ... 
)..enableHttpTracking(
    ignoreUrlPatterns: [RegExp('someUrl/path')],
  );
await DatadogSdk.instance.initialize(datadogConfig, TrackingConsent.granted);
```

### SDK logs

Error:
_TypeError

Message:
Null check operator used on a null value

StackTrace:
```
#0      DatadogTrackingHttpClient._openUrl (package:datadog_tracking_http_client/src/tracking_http_client.dart:142:19)
<asynchronous suspension>
#1      Future.timeout.<anonymous closure> (dart:async/future_impl.dart:940:15)
<asynchronous suspension>
#2      IOHttpClientAdapter._fetch (package:dio/src/adapters/io_adapter.dart:94:19)
<asynchronous suspension>
#3      CancelableCompleter.complete.<anonymous closure> (package:async/src/cancelable_operation.dart:425:16)
<asynchronous suspension>
```

### Expected behavior

`DatadogTrackingHttpClient` should throw original errors.

### Affected SDK versions

datadog_tracking_http_client 2.1.0

### Did you confirm if the latest SDK version fixes the bug?

Yes

### Flutter Version

3.20.0-1.2.pre

**Comment by fuzzybinary** [comment]
Thanks for reporting and taking the time to submit a fix!

Once your PR is in, I'll try to get it into a patch release early next week.

**Comment by fuzzybinary** [comment]
This is now available in `datadog_tracking_http_client`  v2.1.1

**Comment by ronnnnn** [comment]
@fuzzybinary 
Thanks for quick release.
